### PR TITLE
Feature/hook assets

### PR DESCRIPTION
**Pull request by nikkiki**
Signed-off-by: Nikki Attea <nikki@sensu.io>

## What is this change?

Adds the `runtime_assets` field to `HookConfig`. Adds assets to check hook execution.

## Why is this change necessary?

Closes https://github.com/sensu/sensu-go/issues/929

## Does your change need a Changelog entry?

Yas.

## Do you need clarification on anything?

Nah.

## Were there any complications while making this change?

Nah.

## Have you reviewed and updated the documentation for this change? Is new documentation required?

https://github.com/sensu/sensu-docs/issues/1609

## How did you verify this change?

https://sensuinc.testrail.io/index.php?/cases/view/8783&group_by=cases:section_id&group_order=asc&group_id=175
Required by https://github.com/sensu/sensu-staging/pull/144
Required by https://github.com/sensu/sensu-go-qa-crucible/pull/61

### CODE REVIEW

**Comment by ccressent** [comment]
I accidentally deleted @nikkiki's branch, which closed her original PR...
She had to open this new one instead. The previous discussion is [here](https://github.com/sensu/sensu-go/pull/3134).

File: backend/schedulerd/executor.go [hunk 1]
```diff
@@ -355,6 +354,13 @@ func buildRequest(check *types.CheckConfig, s store.Store) (*types.CheckRequest,
 		for _, hook := range hooks {
 			if hookIsRelevant(hook, check) {
 				request.Hooks = append(request.Hooks, *hook)
+				if len(hook.RuntimeAssets) != 0 {
+					for _, asset := range assets {
+						if assetIsRelevant(asset, hook.RuntimeAssets) {
+							request.Assets = append(request.Assets, *asset)
+						}
+					}
+				}
```
Comment (jamesdphillips) [inline backend/schedulerd/executor.go]: Could flattening all of our assets lead to some unexpected behaviour? 

as an example my check defines `RuntimeAssets = ["python2"]` and one of my hooks defines `RuntimeAssets = ["python3"]`

Comment (nikkiki) [reply backend/schedulerd/executor.go]: Both of those assets would be loaded into the check request, however the execution environment for the check and hook should be different.

Comment (nikkiki) [reply backend/schedulerd/executor.go]: Actually, there is some logic I can add to ensure this!

Comment (nikkiki) [reply backend/schedulerd/executor.go]: @jamesdphillips I added a map `HookAssets` to the `CheckRequest` to ensure only the assets required for each check and hook are loaded into the environment. I'd love to get a gut check on that change!

**Pull request merged**